src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java

```diff
@@ -118,6 +118,7 @@ public synchronized void commit() throws BackendException {
                 failing = false;
                 break;
             } catch (IllegalStateException | ExecutionException e) {
+                log.warn("failed to commit transaction", e);
                 if (isolationLevel.equals(IsolationLevel.SERIALIZABLE) ||
                         isolationLevel.equals(IsolationLevel.READ_COMMITTED_NO_WRITE)) {
```

Comment:
for strong consistency, isolation levels should not be changed dynamically. favouring the existing code as the errors may justify the use case.